### PR TITLE
Update to 40.0

### PR DESCRIPTION
**Pull request by A6GibKm**
Not working due to some tracker issue

```
flatpak run org.gnome.Photos//master

(org.gnome.Photos:2): Tracker-CRITICAL **: 10:20:30.110: Could not create proxy on the D-Bus session bus, Error calling StartServiceByName for org.gnome.Photos.Tracker1.Miner.Files: Timeout was reached

(org.gnome.Photos:2): Tracker-CRITICAL **: 10:20:55.136: Could not create proxy on the D-Bus session bus, Error calling StartServiceByName for org.gnome.Photos.Tracker1.Miner.Extract: Timeout was reached

(org.gnome.Photos:2): Tracker-CRITICAL **: 10:21:20.161: Could not create proxy on the D-Bus session bus, Error calling StartServiceByName for org.gnome.Photos.Tracker1.Miner.Files: Timeout was reached

(org.gnome.Photos:2): Tracker-CRITICAL **: 10:21:45.201: Could not create proxy on the D-Bus session bus, Error calling StartServiceByName for org.gnome.Photos.Tracker1.Miner.Files: Timeout was reached

(org.gnome.Photos:2): Tracker-CRITICAL **: 10:22:10.227: Could not create proxy on the D-Bus session bus, Error calling StartServiceByName for org.gnome.Photos.Tracker1.Miner.Extract: Timeout was reached

(org.gnome.Photos:2): Tracker-CRITICAL **: 10:22:35.253: Could not create proxy on the D-Bus session bus, Error calling StartServiceByName for org.gnome.Photos.Tracker1.Miner.Files: Timeout was reached

(org.gnome.Photos:2): Tracker-CRITICAL **: 10:23:00.311: Could not create proxy on the D-Bus session bus, Error calling StartServiceByName for org.gnome.Photos.Tracker1.Miner.Files: Timeout was reached

(org.gnome.Photos:2): Tracker-CRITICAL **: 10:23:25.321: Could not create proxy on the D-Bus session bus, Error calling StartServiceByName for org.gnome.Photos.Tracker1.Miner.Extract: Timeout was reached

(org.gnome.Photos:2): Tracker-CRITICAL **: 10:23:50.347: Could not create proxy on the D-Bus session bus, Error calling StartServiceByName for org.gnome.Photos.Tracker1.Miner.Files: Timeout was reached

(org.gnome.Photos:2): gnome-photos-WARNING **: 10:23:50.349: Unable to activate miner org.gnome.Photos.Tracker1.Miner.Files

(org.gnome.Photos:2): gnome-photos-WARNING **: 10:23:50.435: Unable to fetch the list of available renderers: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
```

It ultimately opens, but after a non acceptable amount of time.

### CODE REVIEW

**Comment by A6GibKm** [comment]
On my desktop machine on fedora 32 I get on first run
```
(org.gnome.Photos:2): gnome-photos-WARNING **: 14:20:12.221: Unable to fetch the list of available renderers: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable

(org.gnome.Photos:2): Tracker-CRITICAL **: 14:20:14.331: tracker_sparql_cursor_real_get_boolean: assertion '_tmp0_ == TRACKER_SPARQL_VALUE_TYPE_BOOLEAN' failed

(org.gnome.Photos:2): Tracker-CRITICAL **: 14:20:26.993: tracker_sparql_cursor_real_get_boolean: assertion '_tmp0_ == TRACKER_SPARQL_VALUE_TYPE_BOOLEAN' failed

....
```

Then it works as expected with only  the first warning.

The previous reports were on Fedora 33.

File: org.gnome.Photos.json [hunk 1]
```diff
@@ -103,22 +104,11 @@
                 }
             ]
         },
-        {
```
Comment (bilelmoussaoui) [inline org.gnome.Photos.json:106]: Why this change? especially that you re-add it later.

File: org.gnome.Photos.json [hunk 2]
```diff
@@ -252,55 +254,51 @@
...
             "buildsystem": "meson",
-            "cleanup": [ "/bin", "/etc", "/lib/girepository-1.0", "/libexec", "/share/dbus-1", "/share/gir-1.0" ],
+            "cleanup": [ "/bin", "/etc", "/lib/girepository-1.0", "/share/dbus-1", "/share/gir-1.0" ],
             "config-opts": [ "-Dbash_completion=no", "-Ddocs=false", "-Dsystemd_user_services=no", "-Dman=false" ],
             "sources": [
                 {
                     "type": "archive",
-                    "url": "https://download.gnome.org/sources/tracker/2.3/tracker-2.3.4.tar.xz",
-                    "sha256": "577952244ab977c78b0b88e2f63c4197eaba16e4d66bff692b7f58993e06516d"
+                    "url": "https://download.gnome.org/sources/tracker/2.3/tracker-2.3.6.tar.xz",
```
Comment (bilelmoussaoui) [inline org.gnome.Photos.json]: Photos 3.38 uses tracker3 iirc.

File: org.gnome.Photos.json (hunk 1 above)
Comment (A6GibKm) [reply org.gnome.Photos.json:106]: Based on https://gitlab.gnome.org/GNOME/gnome-photos/-/commit/bfbf250252bb93db02a9866f3c1468e9b2386a11. I don't think it changes much, but why risk it.

File: org.gnome.Photos.json (hunk 2 above)
Comment (A6GibKm) [reply org.gnome.Photos.json]: It is a work in progress https://gitlab.gnome.org/GNOME/gnome-photos/-/merge_requests/135.

**Comment by A6GibKm** [comment]
@Eonfge, could you please help me testing https://github.com/flathub/org.gnome.Photos/pull/14 it builds but it is a little buggy at least on one of my machines.

**Comment by Eonfge** [comment]
## Host system
https://linux-hardware.org/?probe=09a02746ac

## Quick testing

Things I see that might be unrelated to Flatpak:
- right-click to organize photos is totally unclear

Things that could be Flatpak related:
- Under devices, it tells me that I have a network allocated storage and a second HDD, but I can't access either of them.

But, caveat emptor, I've never used photos so I don't know what is supposed to happen when I click all buttons.

**Comment by A6GibKm** [comment]
Thanks for testing, my issue was that sometimes on first run it wont find any media.

**Comment by ghost** [comment]
I see the same issue of no media on first run.

A good way to debug is run the app under `dbus-run-session` so we see all of the log output from autostarted DBus services.

This seems to change the behaviour so that media appears more quickly. I'm not sure why.

I also see warnings like this:

```
 (org.gnome.Photos:2): Tracker-CRITICAL **: 01:20:56.653: tracker_sparql_cursor_real_get_boolean: assertion '_tmp0_ == TRACKER_SPARQL_VALUE_TYPE_BOOLEAN' failed

(org.gnome.Photos:2): Tracker-CRITICAL **: 01:20:56.653: tracker_sparql_cursor_real_get_integer: assertion 'tracker_sparql_cursor_get_value_type (self, column) == TRACKER_SPARQL_VALUE_TYPE_INTEGER' failed
```

**Comment by A6GibKm** [comment]
@garnacho I suppose the fix you mentioned at https://github.com/flathub/org.gnome.Music/pull/24 does not apply here right? (Seems to be a different issue on Tracker1)

**Comment by A6GibKm** [comment]
bot, build org.gnome.Photos

**Comment by garnacho** [comment]
> @garnacho I suppose the fix you mentioned at [flathub/org.gnome.Music#24](https://github.com/flathub/org.gnome.Music/pull/24) does not apply here right? (Seems to be a different issue on Tracker1)

Sorry, missed this question, better late than never I guess... Yes, that's indeed different issue/versions :). Those warnings don't ring any bell though.

**Comment by bilelmoussaoui** [comment]
What's the status of this one? :) is the tracker issue still a thing?

bot, build

**Comment by ghost** [comment]
The Tracker 3 branch is under review now and will hopefully land for GNOME 40: https://gitlab.gnome.org/GNOME/gnome-photos/-/merge_requests/135

**Comment by bilelmoussaoui** [comment]
Any change to update this to 40? :-)

**Comment by A6GibKm** [comment]
@ssssam  Could you please review the latest commit? I am not familiarized with tracker3 (or tracker at all).

**Comment by A6GibKm** [comment]
I am keeping the draft status as long as the weird headerbar is still weird. 
image

**Comment by bilelmoussaoui** [comment]
> I am keeping the draft status as long as the weird headerbar is still weird.
> image

Probably needs a `valign = start`? =)

File: org.gnome.Photos.json [hunk 3]
```diff
@@ -145,8 +135,8 @@
             "sources": [
                 {
                     "type": "archive",
-                    "url": "https://www.libraw.org/data/LibRaw-0.19.5.tar.gz",
-                    "sha256": "40a262d7cc71702711a0faec106118ee004f86c86cc228281d12d16da03e02f5"
+                    "url": "https://www.libraw.org/data/LibRaw-0.20.2.tar.gz",
+                    "sha256": "dc1b486c2003435733043e4e05273477326e51c3ea554c6864a4eafaff1004a6"
                 },
                 {
                     "type": "patch",
```
Comment (bilelmoussaoui) [inline org.gnome.Photos.json]: You should drop the patch here as well

**Comment by A6GibKm** [comment]
bot, build org.gnome.Photos

**Comment by A6GibKm** [comment]
bot, build org.gnome.Photos

File: org.gnome.Photos.json [hunk 4]
```diff
@@ -1,25 +1,26 @@
...
         "--talk-name=org.freedesktop.FileManager1",
-        "--talk-name=org.freedesktop.Tracker1",
-        "--talk-name=org.freedesktop.Tracker1.Miner.Extract",
+        "--talk-name=org.freedesktop.Tracker3.Miner.Files.Control",
         "--talk-name=com.intel.dleyna-renderer",
         "--talk-name=org.gnome.ControlCenter",
         "--talk-name=org.gnome.SettingsDaemon",
         "--talk-name=org.gnome.SettingsDaemon.Housekeeping",
+        "--filesystem=xdg-run/gvfsd",
         "--talk-name=org.gtk.vfs",
```
Comment (bilelmoussaoui) [inline org.gnome.Photos.json]:
```suggestion
        "--filesystem=xdg-run/gvfsd",
        "--filesystem=xdg-run/gvfs",
```

**Comment by A6GibKm** [comment]
bot, build org.gnome.Photos

**Comment by bilelmoussaoui** [comment]
Seems to be working fine, thanks @A6GibKm for taking care of this one!

**Pull request merged**